libs/table/src/lib/directives/tree-grid/tree-grid.cell.directive.ts

```diff
@@ -0,0 +1,179 @@
+	 */
+	@HostListener('keydown.ArrowRight') public moveRight(): void {
+		this.moveToCell(this.ngxTreeGridCell + 1, 'current');
+	}
+
+	/**
+	 * Set focus on the cell above
+	 */
+	@HostListener('keydown.ArrowUp') public moveUp(): void {
+		this.moveToCell(this.ngxTreeGridCell, 'above');
+	}
+
+	/**
+	 * Set focus on the cell below
+	 */
+	@HostListener('keydown.ArrowDown') public moveDown(): void {
+		this.moveToCell(this.ngxTreeGridCell, 'below');
+	}
+
+	/**
+	 * Set focus on the first cell of the grid
+	 */
+	@HostListener('keydown.PageUp') moveToFirstCellOfGrid(): void {
+		this.moveToCell('first', 'first');
+	}
+
+	/**
+	 * Set focus on the first cell of the row
+	 */
+	@HostListener('keydown.Home') moveToFirstOfRow(): void {
+		this.moveToCell('first', 'current');
+	}
+
+	/**
+	 * Set focus on the first cell of the same column of the grid
+	 */
+	@HostListener('keydown.control.Home') moveToFirstColumnOfGrid(): void {
+		this.moveToCell(this.ngxTreeGridCell, 'first');
+	}
+
+	/**
+	 * Set focus on the last cell of the grid
+	 */
+	@HostListener('keydown.PageDown') moveToLastCellOfGrid(): void {
+		this.moveToCell('last', 'last');
+	}
+
+	/**
+	 * Set focus on the last cell of the row
+	 */
+	@HostListener('keydown.End') moveToBottomEnd(): void {
+		this.moveToCell('last', 'current');
+	}
+
+	/**
+	 * Set focus on the last cell of the same column of the grid
+	 */
+	@HostListener('keydown.control.End') moveToBottomControlEnd(): void {
+		this.moveToCell(this.ngxTreeGridCell, 'last');
+	}
+
+	/**
+	 * The index of the cell in the row
+	 */
+	@Input({ required: true }) public ngxTreeGridCell: number;
+
+	/**
+	 * The index of the row
+	 */
+	@Input({ required: true }) public ngxTreeGridCellRow: number;
+
+	constructor(
+		@Optional() private readonly parent: NgxTreeGridDirective,
+		private readonly elementRef: ElementRef
+	) {
+		super();
+	}
+
+	/**
+	 * Sets focus on the cell or on the first focusable item in the cell
+	 */
+	public focus(): void {
+		// Iben: Check if any of the child elements are focusable
+		const focusableElement = this.findFocusableElement();
+
+		// Iben: If no element was focusable, focus on the current element
+		if (!focusableElement) {
+			this.elementRef.nativeElement.focus();
+		}
+	}
+
+	/**
+	 * Moves focus to a provided cell in a provided row
+	 *
+	 * @private
+	 * @param cell - The cell we wish to put focus on
+	 * @param row - The row in which the cell is
+	 */
+	private moveToCell(cell: NgxTreeGridCellTarget, row: NgxTreeGridRowTarget) {
+		this.handleWhenFocussed(() => {
+			this.row.getCell(cell, row)?.focus();
+		});
+	}
+
+	/**
+	 * Searches for a focusable element in the cell
+	 */
+	private findFocusableElement(): HTMLElement | undefined {
+		let result: HTMLElement;
+
+		// Iben: Loop over each first-level element of the children
+		for (let element of [...this.elementRef.nativeElement.children]) {
+			if (!result) {
+				// Iben: Check if we can focus on the element
+				element.focus();
```

Comment:
Does something need to happen if we can't? Just asking because you speak of "check if we can".

Reply:
If we can't then we focus on the main cell in the other method!